ddtrace/tracer/writer.go

```diff
@@ -81,6 +81,8 @@ func (h *agentTraceWriter) flush() {
 	}
 	h.wg.Add(1)
 	h.climit <- struct{}{}
+	oldp := h.payload
+	h.payload = newPayload()
```

Comment:
I'm fine with this change, but curious about why it's being made. Is there some race?

Reply:
Not that I know of, but the previous version didn't seem right to me. It seemed as if we're using `h.payload` (by passing it to the goroutine's function argument) before assigning a new value to it. It didn't feel right and a bit racey. Perhaps I'm just overly cautious.